### ISSUE TITLE
`unique_consecutive()` is not intuitively understandable at all.

### ISSUE THREAD

**Issue by hyperkai** [pytorch/pytorch#130218]
### 📚 The doc issue

[The doc](https://pytorch.org/docs/stable/generated/torch.unique_consecutive.html) of `unique_consecutive()` says that `return_inverse=True` returns the indices for where elements in the original input ended up in the returned unique list as shown below:

> - input ([Tensor](https://pytorch.org/docs/stable/tensors.html#torch.Tensor)) – the input tensor

> - return_inverse ([bool](https://docs.python.org/3/library/functions.html#bool)) – Whether to also return the indices for where elements in the original input ended up in the returned unique list.

> - return_counts ([bool](https://docs.python.org/3/library/functions.html#bool)) – Whether to also return the counts for each unique element.

> - dim ([int](https://docs.python.org/3/library/functions.html#int)) – the dimension to apply unique. If None, the unique of the flattened input is returned. default: None

But the indices are not returned while `return_counts=True` properly returns the couns for each unique element as shown below:

```python
import torch

my_tensor = torch.tensor([[[2, 2, 0], [0, 1, 1]],
                          [[1, 3, 0], [0, 0, 2]]])
torch.unique_consecutive(input=my_tensor, return_inverse=True, return_counts=True)
#(tensor([2, 0, 1, 3, 0, 2]),
#  tensor([[[0, 0, 1],    ←
#           [1, 2, 2]],   ←
#          [[2, 3, 4],    ←
#           [4, 4, 5]]]), ←
#  tensor([2, 2, 3, 1, 3, 1]))
```

### Suggest a potential alternative/fix

```python
torch.__version__

torch.__version__ # 2.3.0+cu121
```

**Comment by albanD** [comment]
Same question as https://github.com/pytorch/pytorch/issues/130217
Let's move the discussion there to avoid duplicates
